### PR TITLE
Doctests for `RMSE` and `MeanPairwiseDistance`

### PR DESCRIPTION
**Pull request by DevPranjal**
Addresses #2265 

Description: Added doctests for `RMSE` and `MeanPairwiseDistance`

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -341,6 +341,14 @@ def run(self):
...
 
 manual_seed(666)
+
+# create default evaluator for doctests
+
+def process_function(engine, batch):
+    y_pred, y = batch
+    return y_pred, y
+
+default_evaluator = Engine(process_function)
```
Comment (DevPranjal) [inline docs/source/conf.py:351]: I have added a global for the default evaluator to which @vfdev-5 was referring to in [this comment](https://github.com/pytorch/ignite/pull/2287#discussion_r736646796).
I also felt we could mention it somewhere... What could be a right place to mention it, if necessary? 
Thanks :)

Comment (sdesrozis) [reply docs/source/conf.py:351]: @DevPranjal Thank you. I think we already discussed that point, and the idea is to have simple and copy/paste pieces of code, even if a lot of code could be reused.

@vfdev-5 what do you think ?

File: ignite/metrics/mean_pairwise_distance.py [hunk 2]
```diff
@@ -26,6 +26,32 @@ class MeanPairwiseDistance(Metric):
...
+        The output of the engine's ``process_function`` needs to be in the format of
+        ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y, ...}``. If not, ``output_tranform`` can be added
+        to the metric to transform the output into the form expected by the metric.
+
+        ``y_pred`` and ``y`` should have the same shape.
+
+        .. testcode::
+
+            metric = MeanPairwiseDistance(p=4)
+            metric.attach(default_evaluator, 'l4_distance')
```
Comment (sdesrozis) [inline ignite/metrics/mean_pairwise_distance.py:41]:
```suggestion
            metric.attach(default_evaluator, 'mpd')
```
or anything else but

File: docs/source/conf.py (hunk 1 above)
Comment (vfdev-5) [reply docs/source/conf.py:351]: @sdesrozis I do not see the point to make API examples cluttered with other "unrealistic" code just to make them executable.

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @DevPranjal !

File: docs/source/conf.py (hunk 1 above)
Comment (sdesrozis) [reply docs/source/conf.py:351]: The point is the user can't copy/paste the code to test. I thought we would like to provide users with easily reproducible examples.

Anyway, let's choose whether we hide things and let's deploy 😊

Comment (vfdev-5) [reply docs/source/conf.py:351]: I know, but it means for this metric we'll have the following:
```python
import torch
from torch import nn, optim
from ignite.engine import *
from ignite.handlers import *
from ignite.metrics import *
from ignite.utils import *

manual_seed(666)

def process_function(engine, batch):
    y_pred, y = batch
    return y_pred, y

default_evaluator = Engine(process_function)

# --- API example 
metric = MeanPairwiseDistance(p=4)
metric.attach(default_evaluator, 'mpd')
preds = torch.Tensor([
                [1, 2, 4, 1],
                [2, 3, 1, 5],
                [1, 3, 5, 1],
                [1, 5, 1 ,11]
])
target = preds * 0.75
state = default_evaluator.run([[preds, target]])
print(state.metrics['mpd'])
```

You have half of the code snippet of preps.

Comment (sdesrozis) [reply docs/source/conf.py:351]: We are talking about more or less 30 code lines, and using lambda function, we could reduce this. Not sure that it's worth hiding rather than exposing.

Anyway, that's a matter of taste.
